aten/src/TH/generic/THTensorMoreMath.cpp

```diff
@@ -1290,35 +1290,39 @@ void THTensor_(triu)(THTensor *r_, THTensor *t, int64_t k)
   }
 }
 
-void THTensor_(cat)(THTensor *r_, THTensor *ta, THTensor *tb, int dimension)
+void THTensor_(cat)(THTensor *r_, THTensor *ta, THTensor *tb, int dimension, int pad, scalar_t pad_value)
 {
   THTensor* inputs[2];
   inputs[0] = ta;
   inputs[1] = tb;
-  THTensor_(catArray)(r_, inputs, 2, dimension);
+  THTensor_(catArray)(r_, inputs, 2, dimension, pad, pad_value);
 }
 
-void THTensor_(check_shape_except_dim)(THTensor *first, THTensor *second, int dimension);
-inline void THTensor_(check_shape_except_dim)(THTensor *first, THTensor *second, int dimension)
+void THTensor_(check_shape_except_dim)(THTensor *first, THTensor *second, int dimension, scalar_t pad);
```

Comment:
why is pad a scalar_t?  You are checking whether padding is on, not what the value of the padding is.